packages/firestore/src/remote/serializer.ts

```diff
@@ -852,6 +853,68 @@ export function toQueryTarget(
   return result;
 }
 
+export function toRunAggregationQueryRequest(
+  serializer: JsonProtoSerializer,
+  target: Target
+): ProtoRunAggregationQueryRequest {
+  // Dissect the path into parent, collectionId, and optional key filter.
+  const result: ProtoRunAggregationQueryRequest = {
+    structuredAggregationQuery: {
+      aggregations: [
+        //TODO: dynamically add the aggregate fields in future development
+        {
+          count: {
+          },
+          alias:"count_alias"
+        },
+      ],
+      structuredQuery: {}
+    }
+  };
+  const path = target.path;
```

Comment:
Instead of duplicating all of this logic below, which is quite error-prone for the future, just call `toQueryTarget()` and "extract" the structured query and parent from its return value.